lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -1512,6 +1521,27 @@ OpFoldResult Aten__Getitem__DictStrOp::fold(ArrayRef<Attribute> operands) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// Aten__Contains__IntListOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult Aten__Contains__IntListOp::fold(ArrayRef<Attribute> operands) {
+  auto arrayConstruct = getListConstructIfNotModified(l());
+  if (!arrayConstruct)
+    return nullptr;
+
+  auto targetKey = item();
+  for (auto l : arrayConstruct.elements()) {
+    if (l == targetKey)
+      return getI1IntegerAttr(getContext(), true);
+  }
+  for (auto l : arrayConstruct.elements()) {
+    if (typeid(l) != typeid(torch.constant.int))
+      return getI1IntegerAttr(getContext(), false);
+  }
```

Comment:
This does not work for the `false` case. An element in the list not being a constant int does not mean that the `targetKey` is not in the list.

Because checking that something is not in the list requires us to determine that `targetKey` is *truly* not equal to any element in the list, we can only achieve this when we have enough information about the list elements. For this particular op, the information we need is that we know the int value of all elements in the list and `targetKey` at compile time. So here you first need to convert the list of `torch.constant.int`s to C++ integers as well as `targetKey`, then you can just loop through the list and check that `targetKey` is not equal to any of the elements. There are helper functions to get the integers. If you do `git grep 'm_TorchConstantInt'` you should find examples for getting a `torch.constant.int` and a list of `torch.constant.int`s